### DISCUSSION TITLE
Filter concats without macro detour?

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#6177] [Ideas]
I wish concatenations were possible directly in the filter rather having to detour via a macro. The following recurrence is annoying and makes code harder to read:

```
\define concat() $(foo)$bar

...filter="...prefix<concat>..."
```

Instead something like the following would be neat and very legible:

`...filter="...prefix[<foo>bar]..."`

**Comment by pmario** [comment]
>...The following recurrence is annoying and makes code harder to read:

but much more flexible in the long run

**Reply by pmario** [reply]
BUT ... yes ... concatenation should be easier!

**Comment by saqimtiaz** [comment]
See #5121 and also [some 5.2.0 shenanigans](https://talk.tiddlywiki.org/t/5-2-0-new-features-macro-parameters-in-filter/1135/4?u=saqimtiaz)

The challenge is making text substitution easier without adding too much cognitive overload for users in terms of new syntax.
